src/test/java/org/folio/marccat/resources/TenantTest.java

```diff
@@ -1,59 +1,125 @@
+    p2.setValue("false");
+
+    Parameter p3 = new Parameter();
+    p3.setKey("loadBibliographicSample");
+    p3.setValue("false");
+    List<Parameter> parameters = Arrays.asList(p, p2, p3);
+    ta.setParameters(parameters);
+    Map<String, String> headers = addDefaultHeaders(url, StorageTestSuite.TENANT_ID);
+    Response response = given().headers(headers).body(ta).when().post(url);
+
+    String failureMessage = String.format("Tenant init failed: %s: %s", response.getStatusCode(), response.getBody());
+
+    assertThat(failureMessage, response.getStatusCode(), is(201));
+
+  }
 
   @Test
-  public void create()
-    throws InterruptedException,
-    ExecutionException,
-    TimeoutException,
-    MalformedURLException {
+  public void create_withoutLoadData()
+      throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
     String url = getURI("/_/tenant");
-    JSONObject jo = new JSONObject();
-    String moduleFrom = "mod-marccat-2.3.0";
-    String moduleTo ="";
-    if (moduleFrom != null) {
-      jo.put("module_from", moduleFrom);
-    }
-    jo.put("module_to", moduleTo);
+
+    TenantAttributes ta = new TenantAttributes();
+    ta.setModuleFrom("mod-marccat-2.3.0");
+    ta.setModuleTo("");
+
+    Parameter p = new Parameter();
+    p.setKey("loadSample");
+    p.setValue("false");
+
+    Parameter p2 = new Parameter();
+    p2.setKey("loadReference");
+    p2.setValue("false");
+
+    Parameter p3 = new Parameter();
+    p3.setKey("loadBibliographicSample");
+    p3.setValue("false");
+    List<Parameter> parameters = Arrays.asList(p, p2, p3);
+    ta.setParameters(parameters);
+
     Map<String, String> headers = addDefaultHeaders(url, StorageTestSuite.TENANT_ID);
-    Response response = given()
-      .headers(headers)
-      .body(jo)
-      .when()
-      .post(url);
-    //Thread.sleep(120000);
+    Response response = given().headers(headers).body(ta).when().post(url);
 
-    String failureMessage = String.format("Tenant init failed: %s: %s",
-      response.getStatusCode(), response.getBody());
+    String failureMessage = String.format("Tenant init failed: %s: %s", response.getStatusCode(), response.getBody());
 
-    assertThat(failureMessage,
-      response.getStatusCode(), is(201));
+    assertThat(failureMessage, response.getStatusCode(), is(201));
 
   }
 
+  @Test
+  public void create_withLoadReference()
+      throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+    String url = getURI("/_/tenant");
+
+    TenantAttributes ta = new TenantAttributes();
+    ta.setModuleFrom("mod-marccat-2.3.0");
+    ta.setModuleTo("");
+
+    Parameter p = new Parameter();
+    p.setKey("loadSample");
+    p.setValue("true");
+
+    Parameter p2 = new Parameter();
+    p2.setKey("loadReference");
+    p2.setValue("true");
+
+    Parameter p3 = new Parameter();
+    p3.setKey("loadBibliographicSample");
+    p3.setValue("false");
```

Comment:
It seems we don't have a test for "loadBibliographicSample" "true". Can we add one to cover that code path?

Reply:
@carmentrazza the load bibliographic sample doesn't work with test. For the moment, I have controlled that it isn't loaded in the test, but I think it should be fixed in another PR.

Reply:
I have seen that the test fails because it is not possible to make calls to OKAPI. It has to be done in another PR.